### PR TITLE
feat: Improve the FeedbackDataset.__repr__ method

### LINKED ISSUE #3935: [IMPROVEMENT] Improve the `FeedbackDataset.__repr__` method
maybe this can be formatted a bit more nicely with multi_line. Also, I would say thing like the filters and sorts should be added, when present.

_Originally posted by @davidberenstein1957 in https://github.com/argilla-io/argilla/pull/3925#discussion_r1354907617_

### PR DESCRIPTION
**Pull request by kursathalat**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Closes #3935 

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor (change restructuring the codebase without changing functionality)
- [ ] Improvement (change adding some improvement to an existing functionality)

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

- [ ] Test A
- [ ] Test B

**Checklist**

- [ ] I added relevant documentation
- [ ] I followed the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the `CHANGELOG.md` file (See https://keepachangelog.com/)

### CODE REVIEW

**Review by davidberenstein1957 — CHANGES_REQUESTED** [review]
Hi @kursathalat, I am not sure if the current formatting makes it more readable for me, could you try to play around with this a bit more and potentially align with the local `FeedbackDataset` and include the `metadata-properties`. Also, you left a unused import.

File: src/argilla/client/feedback/dataset/local/dataset.py [hunk 1]
```diff
@@ -162,13 +163,21 @@ def records(self) -> List["FeedbackRecord"]:
 
     def __repr__(self) -> str:
         """Returns a string representation of the dataset."""
-        return (
```
Comment (davidberenstein1957) [inline src/argilla/client/feedback/dataset/local/dataset.py:165]: @kursathalat we should just use this old default formatting but add `metadata_properties` to it.

File: src/argilla/client/feedback/dataset/remote/dataset.py [hunk 2]
```diff
@@ -418,18 +418,25 @@ def _question_name_to_id(self) -> Dict[str, "UUID"]:
 
     def __repr__(self) -> str:
         """Returns a string representation of the dataset."""
-        indent = "   "
-        return (
```
Comment (davidberenstein1957) [inline src/argilla/client/feedback/dataset/remote/dataset.py:422]: @kursathalat we should just use the old default formatting as is shown for the `FeedbackDataset` but add `metadata_properties` to it.

Additionally, we need to add some tests.

**Comment by kursathalat** [comment]
The problem with the current case is that when you are in a smaller screen, longer lines are wrapped improperly which looks messy. Here is the old version and new version (both looks proper on a bigger screen). If tabulate or any other library offers proper wrapping, we can iterate over the new one.

Ekran Resmi 2023-11-08 17 37 21
Ekran Resmi 2023-11-08 17 37 34

**Comment by davidberenstein1957** [comment]
closing this in favour of https://github.com/argilla-io/argilla/pull/4192

**Pull request closed without merging**